client.go

```diff
@@ -301,7 +303,7 @@ func (c *Client) unmarshalHTTPJSONBody(resp *http.Response, reqURL string, resul
 	}
 
 	if len(body) == 0 {
-		if result != nil {
+		if result != nil && resp.StatusCode != 204 {
```

Comment:
yeah, so I tried doing something like that initially, but AFAIK there is no such thing as an `undefined` type. I thought of creating a special struct like `EmptyResponse` or something that we would check for instead. What do you think about that?

Reply:
I implemented a `NoContentResponse` type, which I believe accomplishes what you want, may not be the most ideal way but the only one that was apparent with my limited Go skills. LMK what you think.

Reply:
I experimented a bit around and found a way how we can forward nil.

Change the resourceAdd function to accept a pointer to the response, this will allow to pass nil:

```golang
func resourceAdd[Resp any](
	ctx context.Context,
	client *Client,
	path string,
	opts any,
	resp *Resp,
) (*Resp, error) {
	req, err := client.newPostRequest(path, opts)
	if err != nil {
		return nil, err
	}

	if err := client.sendRequest(ctx, req, resp); err != nil {
		return nil, err
	}

	return resp, nil
}
```

The functions that call resourceAdd can still kept short by using named return values, like:
```golang
func (s *PullZoneService) Add(ctx context.Context, opts *PullZoneAddOptions) (resp *PullZone, err error) {
	return resourceAdd(
		ctx,
		s.client,
		"/pullzone",
		opts,
		resp,
	)
}
```

I think it's less complex then introducing the NoContentResponse type + using reflection.

What do you think?

Reply:
It seems to me that this approach would actually require more effort in the long run, as every time you call a `resource*` function you'll need to pass an additional parameter. Whereas, with `NoContentResponse` you only need to opt into that on the specific methods that actual return a `204`. But I'm not a Go expert so I might be missing how your proposal actually works in practice. With that said:

1. Again, I'm a Go noob, so I defer your expertise.
2. You're the boss 😄

Reply:
My latest push implements my last concept. Take a look and if you still want to remove `NoContentResponse` lmk and I can do that.